### ISSUE TITLE
Class durations don't match time slot durations due to rounding

### ISSUE THREAD

**Issue by willgearty** [learning-unlimited/ESP-Website#2696]
We round class durations to two decimal places, but we don't really round anything else to two decimal places, so this can cause problems with scheduling and adding co-teachers in cases where the lengths of timeblocks result in decimals with more than two decimal places (e.g. 50 minutes).

I think one of the biggest offenders is here, where we check whether a teacher can teach a class:
https://github.com/learning-unlimited/ESP-Website/blob/9896724526d6506f81531a8f2f0394aa05bd9eea/esp/esp/program/models/class_.py#L1678-L1707

However, there must be similar problems in the code for the scheduler. The scheduling diagnostics also need to be fixed:
https://github.com/learning-unlimited/ESP-Website/blob/9147b7912921642812fa3fc4bfdbca35eaac9665/esp/esp/program/modules/handlers/schedulingcheckmodule.py#L218-L227

Basically, any time we call `duration` and then make a comparison, we should make sure we are rounding consistently.

Alternatively, a lot of problems arise from calculating the lengths of time blocks, so we could just add rounding here:
https://github.com/learning-unlimited/ESP-Website/blob/9147b7912921642812fa3fc4bfdbca35eaac9665/esp/esp/cal/models.py#L75-L76

We'd then need to make sure we are using that method instead of reinventing the wheel each time we want the duration of an event (e.g. `conflicts()` above).

Reported by NEU.
